File: sample/src/main/java/ru/surfstudio/standard/ui/screen/main/MainScreenConfigurator.java
package ru.surfstudio.standard.ui.screen.main;

import android.app.Activity;
import android.content.Intent;

import dagger.Component;
import ru.surfstudio.android.core.app.dagger.scope.PerScreen;
import ru.surfstudio.android.core.ui.base.dagger.CoreActivityScreenModule;
import ru.surfstudio.android.core.ui.base.screen.configurator.ScreenComponent;
import ru.surfstudio.standard.app.dagger.ActivityComponent;
import ru.surfstudio.standard.ui.base.configurator.ActivityScreenConfigurator;
import ru.surfstudio.standard.ui.base.dagger.ActivityScreenModule;

/**
 * Конфигуратор активити главного экрана
 */
class MainScreenConfigurator extends ActivityScreenConfigurator {
    @PerScreen
    @Component(dependencies = ActivityComponent.class, modules = {ActivityScreenModule.class})
    interface MainScreenComponent extends ScreenComponent<MainActivityView> {
    }

    MainScreenConfigurator(Activity activity, Intent intent) {
        super(activity, intent);
    }

    @Override
    protected ScreenComponent createScreenComponent(ActivityComponent activityComponent,
                                                    ActivityScreenModule activityScreenModule,
                                                    CoreActivityScreenModule coreActivityScreenModule,
                                                    Intent intent) {
        return DaggerMainScreenConfigurator_MainScreenComponent.builder()
                .activityComponent(activityComponent)
                .coreActivityScreenModule(coreActivityScreenModule)
                .activityScreenModule(activityScreenModule)
                .build();
    }

    @Override
    public String getName() {
        return "main";
    }
}

File: template/src/main/java/ru/surfstudio/standard/ui/screen/main/MainScreenConfigurator.kt
package ru.surfstudio.standard.ui.screen.main

import android.content.Intent
import dagger.Component
import dagger.Module
import ru.surfstudio.android.core.mvp.configurator.ScreenComponent
import ru.surfstudio.android.dagger.scope.PerScreen
import ru.surfstudio.standard.ui.base.configurator.ActivityScreenConfigurator
import ru.surfstudio.standard.ui.base.dagger.activity.ActivityComponent
import ru.surfstudio.standard.ui.base.dagger.screen.ActivityScreenModule
import ru.surfstudio.standard.ui.base.dagger.screen.CustomScreenModule

/**
 * Конфигуратор активити главного экрана
 */
internal class MainScreenConfigurator(intent: Intent) : ActivityScreenConfigurator(intent) {
    @PerScreen
    @Component(dependencies = arrayOf(ActivityComponent::class), modules = arrayOf(ActivityScreenModule::class, MainScreenModule::class))
    internal interface MainScreenComponent : ScreenComponent<MainActivityView>

    @Module
    internal class MainScreenModule(route: MainActivityRoute) :
            CustomScreenModule<MainActivityRoute>(route)

    override fun createScreenComponent(activityComponent: ActivityComponent,
                                       activityScreenModule: ActivityScreenModule,
                                       intent: Intent): ScreenComponent<*> {
        return DaggerMainScreenConfigurator_MainScreenComponent.builder()
                .activityComponent(activityComponent)
                .activityScreenModule(activityScreenModule)
                .mainScreenModule(MainScreenModule(MainActivityRoute()))
                .build()
    }
}
